hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,26 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    String database = identifier.namespace().level(0);
+    try {
+      Table table = clients.run(client -> client.getTable(database, identifier.name()));
```

Comment:
There seems to be a API tableExists, can we use it?     cc @pvary 

``` boolean tableExists(String var1, String var2) throws MetaException, TException, UnknownDBException;```

Reply:
yeah that's what I initially intend to use, what I think this return true for either hive or iceberg table, the later `HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, identifier));` rely on instance of HiveTable to ensure it's of iceberg table type by checking its table properties

Reply:
Oh right I forgot it